client-src/elements/datatypes.ts

```diff
@@ -1,12 +1,30 @@
+declare global {
+  interface Window {
+    google: any;
+  }
+}
+
 //chromedash-timeline
 export interface Property {
   [key: string]: any;
   0: number;
   1: string;
 }
 
-declare global {
-  interface Window {
-    google: any;
-  }
+//autolink
+export interface TextRun {
+  content: string;
+  tag?: string;
+  href?: string;
 }
+
+export interface Component {
+  refRegs: RegExp[];
+  replacer: (match: any) => never[] | TextRun[];
+}
+
+//external-reviewers
+export type LabelInfo = {
+  description: string;
+  variant: 'primary' | 'success' | 'neutral' | 'warning' | 'danger';
```

Comment:
I wonder if we can import this from Shoelace instead of redefining it ourselves.

Reply:
i dont see it on shoelace's documentation

Reply:
It's kinda documented in https://shoelace.style/components/badge#properties. The type is `SbBadge['variant']` after you have `import type SbBadge from '@shoelace-style/shoelace/dist/components/badge/badge.js';` in your imports.

We don't need to make this sort of change in these initial PRs, but it's the sort of thing we should try to clean up as we migrate the bulk of the code.